src/core/annotation.js

```diff
@@ -3326,6 +3508,22 @@ class FreeTextAnnotation extends MarkupAnnotation {
     super(parameters);
 
     this.data.annotationType = AnnotationType.FREETEXT;
+    this.setDefaultAppearance(parameters);
+    if (!this.appearance && parameters.isOffscreenCanvasSupported) {
+      const fakeUnicodeFont = new FakeUnicodeFont(
+        parameters.xref,
+        "sans-serif"
+      );
+      const fontData = this.data.defaultAppearanceData;
+      this.appearance = fakeUnicodeFont.createAppearance(
+        this._contents.str,
+        this.rectangle,
+        this.rotation,
+        fontData.fontSize || 10,
+        fontData.fontColor
+      );
+      this._streams.push(this.appearance, FakeUnicodeFont.toUnicodeStream);
+    }
```

Comment:
Would it make sense to warn when OffscreenCanvas isn't available, to make it clear why it's potentially not working correctly?
```suggestion
    } else if (!this._isOffscreenCanvasSupported) {
      warn("FreeTextAnnotation: OffscreenCanvas is not supported, annotation may not render correctly.");
    }
```